examples/rex_planar_quadrotor_lqr.jl

```diff
@@ -135,54 +143,348 @@ n0, = RD.dims(model_real)
+Qmpc = copy(Qlqr)
+Rmpc = copy(Rlqr)
+Qfmpc = 100*Qmpc
+
+Qmpc_lifted = Diagonal([ρ; diag(Qmpc); fill(ρ, length(ye)-7)])
+Qfmpc_lifted = Diagonal([ρ; diag(Qfmpc); fill(ρ, length(ye)-7)])
+
+# Nominal MPC controller
+mpc_nominal = TrackingMPC(dmodel_nom, 
+    X_ref, U_ref, Vector(T_ref), Qmpc, Rmpc, Qfmpc; Nt=Nt
+)
 
+# Projected MPC controllers
+model_eDMD_projected = EDMD.ProjectedEDMDModel(model_eDMD)
+model_jDMD_projected = EDMD.ProjectedEDMDModel(model_jDMD)
+mpc_eDMD_projected = TrackingMPC(model_eDMD_projected, 
+    X_ref, U_ref, Vector(T_ref), Qmpc, Rmpc, Qfmpc; Nt=Nt
+)
+mpc_jDMD_projected = TrackingMPC(model_jDMD_projected, 
+    X_ref, U_ref, Vector(T_ref), Qmpc, Rmpc, Qfmpc; Nt=Nt
+)
 
+# Lifted MPC controllers
+mpc_eDMD = TrackingMPC(model_eDMD, 
+    Y_ref, U_ref, Vector(T_ref), Qmpc_lifted, Rmpc, Qfmpc_lifted; Nt=Nt, state_error=lifted_state_error
+)
+mpc_jDMD = TrackingMPC(model_jDMD, 
+    Y_ref, U_ref, Vector(T_ref), Qmpc_lifted, Rmpc, Qfmpc_lifted; Nt=Nt, state_error=lifted_state_error
+)
 
+# Run each controller on the same set of initial conditions
+Random.seed!(2)
+x0_sampler = Product([
+    Uniform(-2.0,2.0),
+    Uniform(-2.0,2.0),
+    Uniform(-deg2rad(70),deg2rad(70)),
+    Uniform(-1.0,1.0),
+    Uniform(-1.0,1.0),
+    Uniform(-0.5,0.5)
+])
+x0_test = [rand(x0_sampler) for i = 1:100]
+errors_nominal = sort!(test_initial_conditions(dmodel_real, mpc_nominal, xe, x0_test, t_sim, dt))
+errors_eDMD_projected = sort!(test_initial_conditions(dmodel_real, mpc_eDMD_projected, xe, x0_test, t_sim, dt))
+errors_jDMD_projected = sort!(test_initial_conditions(dmodel_real, mpc_jDMD_projected, xe, x0_test, t_sim, dt))
+errors_eDMD = sort!(test_initial_conditions(dmodel_real, mpc_eDMD, xe, x0_test, t_sim, dt))
+errors_jDMD = sort!(test_initial_conditions(dmodel_real, mpc_jDMD, xe, x0_test, t_sim, dt))
 
-model_eDMD = run_eDMD(X_train, U_train, dt, eigfuns, eigorders, reg=1e-5, name="planar_quadrotor_eDMD")
-model_jDMD = run_jDMD(X_train, U_train, dt, eigfuns, eigorders, dmodel_nom, reg=1e-5, name="planar_quadrotor_jDMD")
\ No newline at end of file
+p_mpc = @pgf Axis(
+    {
+        xmajorgrids,
+        ymajorgrids,
+        xlabel="Percent of samples",
+        # ylabel="Tracking error",
+        legend_cell_align={left},
+        legend_pos="outer north east",
+        ymax=15e-2,
+        xmax=100,
+    },
+    PlotInc({lineopts..., color=color_nominal, style="solid"}, Coordinates(1:100, errors_nominal)),
+    PlotInc({lineopts..., color=color_eDMD, style="solid"}, Coordinates(1:100, errors_eDMD_projected)),
+    PlotInc({lineopts..., color=color_jDMD, style="solid"}, Coordinates(1:100, errors_jDMD_projected)),
+    PlotInc({lineopts..., color=color_eDMD, style="dashed"}, Coordinates(1:100, errors_eDMD)),
+    PlotInc({lineopts..., color=color_jDMD, style="dashed"}, Coordinates(1:100, errors_jDMD)),
+    Legend(["nominal", "eDMD (projected)", "jDMD (projected)", "eDMD (lifted)", "jDMD (lifted)"])
+)
+# display(p_mpc)
```

Comment:
there's a lot of results being calculated here. it's probably a good idea to save them to a results file so that we can go back and tweak plots without re-running everything (plus it makes it easy for reviewers/interested readers to generate the plots as well and check our results).